### DISCUSSION TITLE
Question about prefetch

### DISCUSSION THREAD

**Discussion by ChristophP** [reduxjs/redux-toolkit#4103] [Q&A] [answered]
Hi everyone 

I read through docs on prefetching
https://redux-toolkit.js.org/rtk-query/usage/prefetching#trigger-function-behavior

and couldn't 100% understand the difference between dispatching the query initiate action and dispatching the prefetch action.

Is it correct to say that prefetch does NOT add a subscription where the initiate thunk does?
EDIT: _I tried prefetch and I could see a new subscription being added in the dev tools. Is this by design? If so, how does one unsubscribe a subscription that was added from prefetch. I couldn't find a way to get a handle on it?_

Background: I am thinking about using react router v6 loaders and was wondering of the prefetch action would be a good fit to trigger when a route is matched. Since trigger the regar initiate action/thunk needs to be unsubscribed if I ever want that to be cleaned up and it's a bit tricky to do.

Additional question:
Would something like passing `{ subscribe: false }` be a good idea in this scenario (unfortunately I couldn't only find a mention but no good explanation on the `subscribe` option in the docs)? I could imagine something where the loader fetches without adding a subscription as a prefetch since the page that is being loaded will have a `useQuery()` call thus adding a subscription anyway. The benefit being that the loader will run before the render while the hook will run after the render is complete.

Thanks in advance for the clarification.

Related but older: https://github.com/reduxjs/redux-toolkit/discussions/2751

**Comment by EskiMojo14** [comment, marked as answer]
see https://github.com/reduxjs/redux-toolkit/issues/1283 for discussion of prefetch's subscription

**Reply by ChristophP** [reply]
Ah thanks for pointing me there.
Seems like prefetch is still a WIP.
